File: Calendar.cs
// Calendar.cs
// Календарь с праздниками на C#

using System;
using System.Collections.Generic;
using System.Linq;

class Calendar
{
    private const string RESET = "\u001B[0m";
    private const string RED = "\u001B[91m";
    private const string GREEN = "\u001B[92m";
    private const string YELLOW = "\u001B[93m";
    private const string CYAN = "\u001B[96m";
    private const string BOLD = "\u001B[1m";

    private static string Colorize(string text, string color) => color + text + RESET;

    private static readonly Dictionary<string, Dictionary<int, Dictionary<int, string>>> Holidays = new()
    {
        ["ru"] = new()
        {
            [1] = new() { {1, "Новый год"}, {7, "Рождество Христово"} },
            [2] = new() { {23, "День защитника Отечества"} },
            [3] = new() { {8, "Международный женский день"} },
            [5] = new() { {1, "Праздник Весны и Труда"}, {9, "День Победы"} },
            [6] = new() { {12, "День России"} },
            [11] = new() { {4, "День народного единства"} },
        },
        ["us"] = new()
        {
            [1] = new() { {1, "New Year's Day"} },
            [7] = new() { {4, "Independence Day"} },
            [11] = new() { {11, "Veterans Day"} },
            [12] = new() { {25, "Christmas Day"} },
        },
        ["by"] = new()
        {
            [1] = new() { {1, "Новы год"}, {7, "Каляды"} },
            [3] = new() { {8, "Міжнародны жаночы дзень"} },
            [5] = new() { {1, "Дзень працы"}, {9, "Дзень Перамогі"} },
            [7] = new() { {3, "Дзень Незалежнасці"} },
            [11] = new() { {7, "Дзень Кастрычніцкай рэвалюцыі"} },
        }
    };

    private static string GetHoliday(int month, int day, string country)
    {
        if (Holidays.TryGetValue(country, out var c) &&
            c.TryGetValue(month, out var m) &&
            m.TryGetValue(day, out var name))
            return name;
        return null;
    }

    private static void PrintCalendar(int year, int month, string country)
    {
        string[] monthNames = { "Январь", "Февраль", "Март", "Апрель", "Май", "Июнь",
                                "Июль", "Август", "Сентябрь", "Окторябрь", "Ноябрь", "Декабрь" };
        Console.WriteLine($"\n{Colorize($"{monthNames[month-1]} {year}", BOLD + CYAN)}");
        Console.WriteLine(Colorize("Пн Вт Ср Чт Пт Сб Вс", BOLD));
        Console.WriteLine("---------------------");

        DateTime first = new DateTime(year, month, 1);
        int startOffset = ((int)first.DayOfWeek + 6) % 7; // 0-пн, 6-вс
        int daysInMonth = DateTime.DaysInMonth(year, month);

        string[] week = new string[7];
        for (int i = 0; i < startOffset; i++) week[i] = "  ";
        for (int d = 1; d <= daysInMonth; d++)
        {
            DateTime date = new DateTime(year, month, d);
            int weekday = ((int)date.DayOfWeek + 6) % 7; // 0-пн, 6-вс
            bool isWeekend = (weekday == 5 || weekday == 6);
            string holiday = GetHoliday(month, d, country);
            string dayStr;
            if (holiday != null)
                dayStr = Colorize($"{d,2}", RED);
            else if (isWeekend)
                dayStr = Colorize($"{d,2}", GREEN);
            else
                dayStr = $"{d,2}";
            week[weekday] = dayStr;
            if (weekday == 6)
            {
                Console.WriteLine(string.Join(" ", week));
                week = new string[7];
            }
        }
        // остаток
        if (week[0] != null)
        {
            Console.WriteLine(string.Join(" ", week.Select(s => s ?? "  ")));
        }

        // Праздники
        if (Holidays.TryGetValue(country, out var c2) && c2.TryGetValue(month, out var m2))
        {
            Console.WriteLine($"\n{Colorize("Праздники:", BOLD + YELLOW)}");
            foreach (var day in m2.Keys.OrderBy(k => k))
                Console.WriteLine($"  {day,2} – {m2[day]}");
        }
        Console.WriteLine();
    }

    private static void Interactive()
    {
        Console.WriteLine(Colorize("Календарь с праздниками", BOLD + CYAN));
        Console.Write("Введите год (пусто для текущего): ");
        string yearInput = Console.ReadLine().Trim();
        int year = string.IsNullOrEmpty(yearInput) ? DateTime.Now.Year : int.Parse(yearInput);
        Console.Write("Введите месяц (1-12, пусто для текущего): ");
        string monthInput = Console.ReadLine().Trim();
        int month = string.IsNullOrEmpty(monthInput) ? DateTime.Now.Month : int.Parse(monthInput);
        if (month < 1 || month > 12)
        {
            Console.WriteLine("Неверный месяц");
            return;
        }
        Console.Write("Введите код страны (ru/us/by, по умолчанию ru): ");
        string country = Console.ReadLine().Trim();
        if (string.IsNullOrEmpty(country)) country = "ru";
        if (!Holidays.ContainsKey(country))
        {
            Console.WriteLine($"Страна {country} не поддерживается, используем ru");
            country = "ru";
        }
        PrintCalendar(year, month, country);
    }

    static void Main(string[] args)
    {
        if (args.Length >= 2)
        {
            int year = int.Parse(args[0]);
            int month = int.Parse(args[1]);
            string country = args.Length >= 3 ? args[2] : "ru";
            if (month < 1 || month > 12)
            {
                Console.WriteLine("Месяц должен быть от 1 до 12");
                return;
            }
            if (!Holidays.ContainsKey(country))
            {
                Console.WriteLine($"Страна {country} не поддерживается, используем ru");
                country = "ru";
            }
            PrintCalendar(year, month, country);
        }
        else
        {
            Interactive();
        }
    }
}

File: Calendar.java
// Calendar.java
// Календарь с праздниками на Java

import java.io.*;
import java.time.*;
import java.time.format.*;
import java.util.*;

public class Calendar {
    // ANSI-цвета
    private static final String RESET = "\u001B[0m";
    private static final String RED = "\u001B[91m";
    private static final String GREEN = "\u001B[92m";
    private static final String YELLOW = "\u001B[93m";
    private static final String CYAN = "\u001B[96m";
    private static final String BOLD = "\u001B[1m";

    private static String colorize(String text, String color) {
        return color + text + RESET;
    }

    // Праздники
    private static final Map<String, Map<Integer, Map<Integer, String>>> HOLIDAYS = new HashMap<>();
    static {
        Map<Integer, Map<Integer, String>> ru = new HashMap<>();
        ru.put(1, Map.of(1, "Новый год", 7, "Рождество Христово"));
        ru.put(2, Map.of(23, "День защитника Отечества"));
        ru.put(3, Map.of(8, "Международный женский день"));
        ru.put(5, Map.of(1, "Праздник Весны и Труда", 9, "День Победы"));
        ru.put(6, Map.of(12, "День России"));
        ru.put(11, Map.of(4, "День народного единства"));
        HOLIDAYS.put("ru", ru);

        Map<Integer, Map<Integer, String>> us = new HashMap<>();
        us.put(1, Map.of(1, "New Year's Day"));
        us.put(7, Map.of(4, "Independence Day"));
        us.put(11, Map.of(11, "Veterans Day"));
        us.put(12, Map.of(25, "Christmas Day"));
        HOLIDAYS.put("us", us);

        Map<Integer, Map<Integer, String>> by = new HashMap<>();
        by.put(1, Map.of(1, "Новы год", 7, "Каляды"));
        by.put(3, Map.of(8, "Міжнародны жаночы дзень"));
        by.put(5, Map.of(1, "Дзень працы", 9, "Дзень Перамогі"));
        by.put(7, Map.of(3, "Дзень Незалежнасці"));
        by.put(11, Map.of(7, "Дзень Кастрычніцкай рэвалюцыі"));
        HOLIDAYS.put("by", by);
    }

    private static String getHoliday(int month, int day, String country) {
        Map<Integer, Map<Integer, String>> c = HOLIDAYS.get(country);
        if (c == null) return null;
        Map<Integer, String> m = c.get(month);
        if (m == null) return null;
        return m.get(day);
    }

    private static void printCalendar(int year, int month, String country) {
        String[] monthNames = {"Январь", "Февраль", "Март", "Апрель", "Май", "Июнь",
                               "Июль", "Август", "Сентябрь", "Октябрь", "Ноябрь", "Декабрь"};
        System.out.printf("\n%s\n", colorize(monthNames[month-1] + " " + year, BOLD + CYAN));
        System.out.println(colorize("Пн Вт Ср Чт Пт Сб Вс", BOLD));
        System.out.println("---------------------");

        LocalDate first = LocalDate.of(year, month, 1);
        int startOffset = first.getDayOfWeek().getValue() - 1; // 1-пн, 7-вс -> 0-6
        int daysInMonth = first.lengthOfMonth();

        StringBuilder[] week = new StringBuilder[7];
        for (int i = 0; i < startOffset; i++) {
            week[i] = new StringBuilder("  ");
        }
        for (int d = 1; d <= daysInMonth; d++) {
            LocalDate date = LocalDate.of(year, month, d);
            int weekday = date.getDayOfWeek().getValue(); // 1-пн, 7-вс
            boolean isWeekend = (weekday == 6 || weekday == 7);
            String holiday = getHoliday(month, d, country);
            String dayStr;
            if (holiday != null) {
                dayStr = colorize(String.format("%2d", d), RED);
            } else if (isWeekend) {
                dayStr = colorize(String.format("%2d", d), GREEN);
            } else {
                dayStr = String.format("%2d", d);
            }
            week[weekday-1] = new StringBuilder(dayStr);
            if (weekday == 7) {
                for (int i = 0; i < 7; i++) {
                    System.out.print(week[i].toString() + (i < 6 ? " " : ""));
                }
                System.out.println();
                week = new StringBuilder[7];
                for (int i = 0; i < 7; i++) week[i] = new StringBuilder();
            }
        }
        // остаток недели
        boolean printed = false;
        for (int i = 0; i < 7; i++) {
            if (week[i] != null && week[i].length() > 0) {
                if (!printed) { printed = true; }
                System.out.print(week[i].toString() + (i < 6 ? " " : ""));
            } else {
                System.out.print("  " + (i < 6 ? " " : ""));
            }
        }
        if (printed) System.out.println();

        // Праздники
        Map<Integer, String> holidaysThisMonth = HOLIDAYS.get(country).get(month);
        if (holidaysThisMonth != null && !holidaysThisMonth.isEmpty()) {
            System.out.printf("\n%s\n", colorize("Праздники:", BOLD + YELLOW));
            List<Integer> sortedDays = new ArrayList<>(holidaysThisMonth.keySet());
            Collections.sort(sortedDays);
            for (int day : sortedDays) {
                System.out.printf("  %2d – %s\n", day, holidaysThisMonth.get(day));
            }
        }
        System.out.println();
    }

    private static void interactive() throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        System.out.println(colorize("Календарь с праздниками", BOLD + CYAN));
        System.out.print("Введите год (пусто для текущего): ");
        String yearInput = reader.readLine().trim();
        int year = yearInput.isEmpty() ? LocalDate.now().getYear() : Integer.parseInt(yearInput);
        System.out.print("Введите месяц (1-12, пусто для текущего): ");
        String monthInput = reader.readLine().trim();
        int month = monthInput.isEmpty() ? LocalDate.now().getMonthValue() : Integer.parseInt(monthInput);
        if (month < 1 || month > 12) {
            System.out.println("Неверный месяц");
            return;
        }
        System.out.print("Введите код страны (ru/us/by, по умолчанию ru): ");
        String country = reader.readLine().trim();
        if (country.isEmpty()) country = "ru";
        if (!HOLIDAYS.containsKey(country)) {
            System.out.printf("Страна %s не поддерживается, используем ru\n", country);
            country = "ru";
        }
        printCalendar(year, month, country);
    }

    public static void main(String[] args) throws IOException {
        if (args.length >= 2) {
            int year = Integer.parseInt(args[0]);
            int month = Integer.parseInt(args[1]);
            String country = args.length >= 3 ? args[2] : "ru";
            if (month < 1 || month > 12) {
                System.out.println("Месяц должен быть от 1 до 12");
                return;
            }
            if (!HOLIDAYS.containsKey(country)) {
                System.out.printf("Страна %s не поддерживается, используем ru\n", country);
                country = "ru";
            }
            printCalendar(year, month, country);
        } else {
            interactive();
        }
    }
}
